packages/@aws-cdk/aws-codedeploy/lib/deployment-group.ts

```diff
@@ -141,19 +172,84 @@ export class ServerDeploymentGroup extends ServerDeploymentGroupRef {
             assumedBy: new cdk.ServicePrincipal('codedeploy.amazonaws.com'),
             managedPolicyArns: ['arn:aws:iam::aws:policy/service-role/AWSCodeDeployRole'],
         });
+        this._autoScalingGroups = (props && props.autoScalingGroups) || [];
+        this.addCodeDeployAgentInstallUserData(props && props.agentInstallationType);
 
         const resource = new cloudformation.DeploymentGroupResource(this, 'Resource', {
             applicationName: this.application.applicationName,
             deploymentGroupName: props && props.deploymentGroupName,
             serviceRoleArn: this.role.roleArn,
             deploymentConfigName: props && props.deploymentConfig &&
                 props.deploymentConfig.deploymentConfigName,
+            autoScalingGroups: props && props.autoScalingGroups &&
+                props.autoScalingGroups.map(asg => asg.autoScalingGroupName()),
         });
 
         this.deploymentGroupName = resource.ref;
         this.deploymentGroupArn = deploymentGroupName2Arn(this.application.applicationName,
             this.deploymentGroupName);
     }
+
+    public get autoScalingGroups(): autoscaling.AutoScalingGroup[] {
+        return this._autoScalingGroups.slice();
+    }
+
+    private addCodeDeployAgentInstallUserData(agentInstallationType?: AgentInstallationType): void {
+        if (!agentInstallationType) {
+            return;
+        }
+
+        const stack = cdk.Stack.find(this);
+        const region = stack.requireRegion('Could not determine the region to download the CodeDeploy agent from');
```

Comment:
This is probably controversial... but why wouldn't we use the `AWS::Region` psuedo parameter here? I know it's not _cdk_ like... but are we effectively forcing people into the env model?

Reply:
I agree with @mindstorms6. If it's possible to use `AwsRegion`, then it's usually preferred at these low-layers.

This should work for you:

```ts
const region = (new cdk.AwsRegion()).toString();
```

It will render a "stringified token" that you can naturally embed into other strings. Upon synthesis, these stringified tokens will be converted to `FnJoin` expressions. Magic!

Reply:
Will embedding `AWS::Region` in User Data actually work? I thought CloudFormation only expanded it in "function calls" like `Fn::Join`, etc.?

Reply:
If you use the method I described, it should work. But you should also test it both in a unit test and an integ test